src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs

```diff
@@ -192,7 +194,7 @@ internal sealed class ConnectionViewModel : BindableBase, IConnectionViewModel {
 
             Uri uri;
             var isPathValid = Uri.TryCreate(Path, UriKind.Absolute, out uri);
-            if (string.IsNullOrEmpty(Name)) {
+            if (!IsValidConnectionName(Name)) {
```

Comment:
We should also validate this in broker side, and reject invalid names in some sensible way (i.e. as API error).